packages/backend/src/modules/user/user.service.ts

```diff
@@ -70,4 +83,107 @@ export default class UserService {
+    const user = await this.findByEmail(email);
+    if (!user) {
+      throw new UserInputError(`User with this email does not exist`);
+    }
+
+    const codeEntity = await this.createNewCodeAndDeletePrevious(email, CodeTypes.RESET_PASSWORD);
+
+    await this.sendgrid.send(
+      {
+        from: config.application.sendgrid.senderEmail,
+        to: email,
+        templateId: config.application.sendgrid.resetPasswordTemplateId,
+        dynamicTemplateData: {
+          resetPasswordUrl: `${config.client.host}/reset-password?code=${codeEntity.code}&email=${email}`,
+          firstName: user.firstName || user.email,
+        }
+      }
+    );
+
+    return true;
+  }
+
+  public async resetPassword(email: string, code: string, password: string): Promise<boolean> {
+    const isConfirmed = await this.acceptConfirmationCode(email, CodeTypes.RESET_PASSWORD, code);
+    if (!isConfirmed) throw new UserInputError('Confirmation code was not found');
+
+    const user = await this.findByEmail(email);
+
+    if (!user) {
+      throw new UserInputError(`User with email: ${email} was found`);
+    }
+    user.password = password;
+
+    await user.save();
+
+    return true;
+  }
+
+  public async sendEmailConfirmationCode(email: string): Promise<void> {
+    const user = await this.findByEmail(email);
+    if (!user) {
+      throw new UserInputError(`User with this email does not exist`);
+    }
+
+    const codeEntity = await this.createNewCodeAndDeletePrevious(email, CodeTypes.EMAIL_CONFIRMATION);
+
+    await this.sendgrid.send(
+      {
+        from: config.application.sendgrid.senderEmail,
+        to: email,
+        templateId: config.application.sendgrid.confirmationTemplateId,
+        dynamicTemplateData: {
+          confirmationUrl: `${config.application.host}/user/confirm-email?code=${codeEntity.code}&email=${email}`,
+          firstName: user.firstName || email,
+        }
+      }
+    );
+  }
+
+  public async isEmailNotConfirmed(email: string): Promise<boolean> {
+    const code = await this.codeModel.findOne({
+      email,
+      type: CodeTypes.EMAIL_CONFIRMATION
+    });
+    return Boolean(code);
+  }
+
+  private async createNewCodeAndDeletePrevious(email: string, type: CodeTypes): Promise<IConfirmationCodeDocument> {
+    await this.codeModel.deleteMany({
+      email,
+      type
+    });
+
+    return this.codeModel.create({
+      email,
+      type,
+      expiredAt: new Date(Date.now() + CodeExpirationTime[type])
+    });
+  }
+
+  public async acceptConfirmationCode(email: string, type: CodeTypes, code: string): Promise<boolean> {
+    const context = `${this.constructor.name}:${this.acceptConfirmationCode.name}`;
+    this.logger.debug('Looking for confirmation code', context, { email, code, type });
+    const confirmationCode = await this.codeModel.findOne({
+      code,
+      email,
+      type,
+      expiredAt: {
+        $gte: new Date(),
+      }
+    });
```

Comment:
ESLint warning, semicolon is not needed.

Reply:
I don't see any warnings, prettier puts it here as it the end of statement.

Reply:
Bizarre, the comment is on the wrong line for some reason. I'll close this one and will add another one.